### ISSUE TITLE
The flavor of the build is duplicated in the name of .xctestrun file when running tests on iOS

### ISSUE THREAD

**Issue by Wreos** [leancodepl/patrol#1586]
### Steps to reproduce

I have the following problem with iOS tests. The flavor prefix in the name of .xctest run file is duplicated when I run tests on ios and Patrol is not able to find it
It happens because Patrol expects to see something like

`.xctestrun matching qa_iphonesimulator16.4*.xctestrun`

But somehow flavor prefix is being duplicated when I run  **patrol test --target integration_test/main_test.dart**

`qa_qa_iphonesimulator16.4-arm64-x86_64.xctestrun`

**Generated files**
- Debug-qa-iphonesimulator
- qa_qa_iphonesimulator16.4-arm64-x86_64.xctestrun

**Build configuration**
![image](https://github.com/leancodepl/patrol/assets/15965366/e3c14f8d-900f-476d-bf18-25a9aa626f07)

Pubspecs.yaml
```
patrol:
  app_name: App QA
  android:
    flavor: qa
    package_name: de.app.customerApp
  ios:
    flavor: qa
    bundle_id: de.app.customerApp
```

### Actual results

As result, patrol is not able to find xctestrun file

```
Looking for .xctestrun matching qa_iphonesimulator16.4*.xctestrun at /Users/aleksandrlozhkovoi/Development/CustomerApp/build/ios_integ/Build/Products
Error: No .xctestrun file was found at /Users/aleksandrlozhkovoi/Development/CustomerApp/build/ios_integ/Build/Products
```

### Logs

<details>
<summary>Logs</summary>

```
aleksandrlozhkovoi@Aleksandrs-MacBook-Pro CustomerApp % patrol test --target integration_test/main_test.dart --verbose
You're using Patrol CLI 2.0, which has breaking changes.
Read the migration guide at https://patrol.leancode.co/v2.
Disable this warning by setting the PATROL_MIGRATED environment variable.

Verbose mode enabled. More logs will be printed.
Received 1 test target(s)
Received test target: /Users/aleksandrlozhkovoi/Development/CustomerApp/integration_test/main_test.dart
Generated entrypoint /Users/aleksandrlozhkovoi/Development/CustomerApp/integration_test/test_bundle.dart with 1 bundled test(s)
Received Android flavor: qa
Received iOS flavor: qa
$ flutter --no-version-check devices --machine
No device specified, using the first one (iPhone 14)
Received 1 device(s) to run on
Received device: iPhone 14
Received 7 --dart-define(s) (0 custom, 7 internal)
Received internal --dart-define: PATROL_WAIT=0
Received internal --dart-define: PATROL_APP_PACKAGE_NAME=de.App.customerApp
Received internal --dart-define: PATROL_APP_BUNDLE_ID=de.App.customerApp
Received internal --dart-define: PATROL_ANDROID_APP_NAME=App QA
Received internal --dart-define: PATROL_IOS_APP_NAME=App QA
Received internal --dart-define: INTEGRATION_TEST_SHOULD_REPORT_RESULTS_TO_NATIVE=false
Received internal --dart-define: PATROL_TEST_LABEL_ENABLED=true
• Building app with entrypoint test_bundle.dart for iOS simulator (debug)...
$ flutter build ios --no-version-check --config-only --no-codesign --debug --simulator --flavor qa --target /Users/aleksandrlozhkovoi/Development/CustomerApp/integration_test/test_bundle.dart --dart-define PATROL_WAIT=0 --dart-define PATROL_APP_PACKAGE_NAME=de.App.customerApp --dart-define PATROL_APP_BUNDLE_ID=de.App.customerApp --dart-define PATROL_ANDROID_APP_NAME=App QA --dart-define PATROL_IOS_APP_NAME=App QA --dart-define INTEGRATION_TEST_SHOULD_REPORT_RESULTS_TO_NATIVE=false --dart-define PATROL_TEST_LABEL_ENABLED=true
	Building de.App.customerApp.qa for simulator (ios)...
$ xcodebuild build-for-testing -workspace Runner.xcworkspace -scheme qa -configuration Debug-qa -only-testing RunnerUITests -sdk iphonesimulator -destination generic/platform=iOS Simulator -quiet -derivedDataPath ../build/ios_integ OTHER_SWIFT_FLAGS=$(inherited) -D PATROL_ENABLED
	2023-07-27 12:49:42.179 xcodebuild[50387:9000272] DVTCoreDeviceEnabledState: DVTCoreDeviceEnabledState_Disabled set via user default (DVTEnableCoreDevice=disabled)
	/Users/aleksandrlozhkovoi/Development/CustomerApp/ios/Runner/AppDelegate.swift:18:58: warning: conditional cast from 'AppDelegate' to 'any UNUserNotificationCenterDelegate' always succeeds
	      UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
	                                                         ^
	/Users/aleksandrlozhkovoi/Development/CustomerApp/ios/Runner/AppDelegate.swift:18:58: warning: conditional cast from 'AppDelegate' to 'any UNUserNotificationCenterDelegate' always succeeds
	      UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
	                                                         ^
	note: Building targets in dependency order
	note: Run script build phase 'Run Script' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'Runner' from project 'Runner')
	note: Run script build phase 'Copy GoogleService-Info file to the right location' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'Runner' from project 'Runner')
	note: Run script build phase 'Thin Binary' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'Runner' from project 'Runner')
	warning: Run script build phase 'Run Script' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'Runner' from project 'Runner')
	warning: Run script build phase 'Firebase config' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'Runner' from project 'Runner')
	warning: Run script build phase 'xcode_backend build' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'RunnerUITests' from project 'Runner')
	warning: Run script build phase 'xcode_backend embed_and_thin' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'RunnerUITests' from project 'Runner')
✓ Completed building app with entrypoint test_bundle.dart for iOS simulator (65.8s)
Will uninstall apps before running tests
$ xcrun simctl uninstall C1770B26-B419-4C82-834B-D9294B9B089D de.App.customerApp
$ xcrun simctl uninstall C1770B26-B419-4C82-834B-D9294B9B089D de.App.customerApp.RunnerUITests.xctrunner
• Running app with entrypoint test_bundle.dart for iOS simulator on simulator iPhone 14...
$ xcodebuild -showsdks -json
Assuming SDK version 16.4 for iphonesimulator
Looking for .xctestrun matching qa_iphonesimulator16.4*.xctestrun at /Users/aleksandrlozhkovoi/Development/CustomerApp/build/ios_integ/Build/Products
Error: No .xctestrun file was found at /Users/aleksandrlozhkovoi/Development/CustomerApp/build/ios_integ/Build/Products
#0      throwToolExit (package:patrol_cli/src/base/exceptions.dart:7:3)
#1      IOSTestBackend.xcTestRunPath (package:patrol_cli/src/ios/ios_test_backend.dart:243:7)
<asynchronous suspension>
#2      IOSTestBackend.execute.<anonymous closure> (package:patrol_cli/src/ios/ios_test_backend.dart:155:26)
<asynchronous suspension>
#3      DisposeScope.run (package:dispose_scope/src/dispose_scope.dart:46:7)
<asynchronous suspension>
#4      IOSTestBackend.execute (package:patrol_cli/src/ios/ios_test_backend.dart:147:5)
<asynchronous suspension>
#5      TestCommand._execute (package:patrol_cli/src/commands/test.dart:269:7)
<asynchronous suspension>
#6      TestCommand.run (package:patrol_cli/src/commands/test.dart:169:23)
<asynchronous suspension>
#7      CommandRunner.runCommand (package:args/command_runner.dart:212:13)
<asynchronous suspension>
#8      PatrolCommandRunner.runCommand (package:patrol_cli/src/runner/patrol_command_runner.dart:310:18)
<asynchronous suspension>
#9      PatrolCommandRunner.run (package:patrol_cli/src/runner/patrol_command_runner.dart:254:18)
<asynchronous suspension>
#10     patrolCommandRunner (package:patrol_cli/src/runner/patrol_command_runner.dart:72:20)
<asynchronous suspension>
#11     main (file:///Users/aleksandrlozhkovoi/.pub-cache/hosted/pub.dev/patrol_cli-2.0.4/bin/main.dart:6:20)
<asynchronous suspension>

```

</details>

### Patrol version

[patrol 2.1.0](patrol: ^2.1.0)

### Patrol Doctor output

<details>
<summary>Patrol Doctor output</summary>

```
Patrol CLI version: 2.0.4
Program adb found in /opt/homebrew/bin/adb
Env var $ANDROID_HOME is not set
Program xcodebuild found in /usr/bin/xcodebuild
Program ideviceinstaller found in /opt/homebrew/bin/ideviceinstaller
```

</details>

### Flutter Doctor output

<details>
<summary>Flutter Doctor output</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.10.5, on macOS 13.4.1 22F82 darwin-arm64, locale
    en-DE)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.2)
[✓] VS Code (version 1.80.1)
[✓] Connected device (4 available)
[✓] Network resources
```

</details>

**Comment by jBorkowska** [comment]
Hi @Wreos. Thank you for posting the issue.

It may be related to the way you did Patrol-related setup with flavours. Please try to remove the flavor definition from Patrol section in pubspec.yaml. Let me know if the issue persists

**Comment by Wreos** [comment]
Hi @jBorkowska. Big thanks for the answer. I tried to remove flavor from pubspecs, but as app uses flavours I have to provide it anyway. So I i passed flavour as parameter **patrol test --target integration_test --qa**    
but still have this error, the prefix is duplicated

**Comment by chris-wickens** [comment]
I'm running into the same issue following the setup tutorial, Android tests run fine but I'm unable to prevent the xctestrun file including the flavour name twice, e.g. dev_dev_iphonesimulator16.4-arm64-x86_64.xctestrun

**Comment by jBorkowska** [comment]
That's really strange, we never encountered it in our projects, where we use flavours as well. @Wreos @chris-wickens Could any of you provide some repo with minimal reproducible example? It would be great for us to investigate there

**Comment by chris-wickens** [comment]
Sorry for the late response, I reproduced this in a new project here: https://github.com/chris-wickens/patrol_config_issue

Running `patrol test --target integration_test/example_test.dart` creates the file `dev_dev_iphonesimulator16.4-arm64-x86_64.xctestrun` in `../patrol_config_issue/build/ios_integ/Build/Products/`.

**Comment by bartekpacia** [comment]
I can reproduce this issue.

```
$ patrol --version
patrol_cli v2.1.0
```

**Comment by jBorkowska** [comment]
Probably the issue is related to iOS 16.4 - I can't reproduce it on 16.2

**Comment by bartekpacia** [comment]
Hi @chris-wickens and @Wreos - are you by any chance using XC Test Plans?

**Comment by chris-wickens** [comment]
No, my project had no native integration tests same as the minimal sample.

**Comment by chris-wickens** [comment]
Just tested and the issue is fixed for me in the latest version, thanks very much!

**Comment by bartekpacia** [comment]
Glad to hear that :D

**Comment by bartekpacia** [comment]
Fixed in #1636

**Comment by Wreos** [comment]
I've just tested Patrol 1.12. It looks that issue has been fixed. Big thanks for the quick fix. :)
